src/dotnet/commands/dotnet-build3/ForwardingApp.cs

```diff
@@ -0,0 +1,104 @@
+﻿using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.ProjectModel;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Cli
+{
+    /**
+     * A class which encapsulates logic needed to forward arguments from the current process to another process
+     * invoked with the dotnet.exe host.
+     */ 
+    public class ForwardingApp
+    {
+        private static readonly string s_hostExe = "dotnet";
+
+        private readonly string _forwardApplicationPath;
+        private readonly string[] _argsToForward;
+        private readonly string _depsFile;
+        private readonly string _runtimeConfig;
+        private readonly string _additionalProbingPath;
+        private readonly Dictionary<string, string> _environment;
+
+        private string[] _allArgs;
+        
+        public ForwardingApp(
+            string forwardApplicationPath, 
+            string[] argsToForward, 
+            string depsFile = null, 
+            string runtimeConfig = null,
+            string additionalProbingPath = null,
+            Dictionary<string, string> environment = null)
+        {
+            _forwardApplicationPath = forwardApplicationPath;
+            _argsToForward = argsToForward;
+            _depsFile = depsFile;
+            _runtimeConfig = runtimeConfig;
+            _additionalProbingPath = additionalProbingPath;
+            _environment = environment;
+            
+            var allArgs = new List<string>();
```

Comment:
See @eerhardts comments on the previous PR. The instance variable needs to be an array. This is essentially constructing the array. 

Open to cleaner approaches here.